### PR TITLE
Issue 5030 (4-2-stable)

### PR DESCRIPTION
**Pull request by JustinKyleJames**

### CODE REVIEW

File: server/api/src/rsDataObjRepl.cpp [hunk 1]
```diff
@@ -424,6 +425,11 @@ int dataObjCopy(
     int srcRemoteFlag{};
     int source_l1descInx = L1desc[_destination_l1descInx].srcL1descInx;
     int srcL3descInx = L1desc[source_l1descInx].l3descInx;
+
+    // copy data size from src catalog to destination
+    L1desc[_destination_l1descInx].dataSize = L1desc[source_l1descInx].dataObjInfo->dataSize;
+    //L1desc[source_l1descInx].dataSize = L1desc[source_l1descInx].dataObjInfo->dataSize;
```
Comment (trel) [inline server/api/src/rsDataObjRepl.cpp:431]: leftover?

Comment (JustinKyleJames) [reply server/api/src/rsDataObjRepl.cpp:431]: Left it in there in case someone decides it is needed in the fruture.  Can delete

**Comment by trel** [comment]
do we need a new issue for the first commit?

File: server/api/src/rsDataObjRepl.cpp (hunk 1 above)
Comment (trel) [reply server/api/src/rsDataObjRepl.cpp:431]: delete it if it's not needed today.   leave a helpful comment if there's something to say.

Comment (JustinKyleJames) [reply server/api/src/rsDataObjRepl.cpp:431]: Probably just left over.

**Comment by trel** [comment]
we need any tests for this?   handled by existing tests elsewhere?

**Comment by JustinKyleJames** [comment]
For 5078, the new S3 plugin won't work without it so it would be tested with the plugin itself.  Not sure how to test it in the core itself.

For 4030, the resource plugin has to set that.  If it is tested it would be in the plugin.

**Comment by trel** [comment]
Got it - thanks.

**Comment by trel** [comment]
If we're happy with this, please # it.

**Comment by JustinKyleJames** [comment]
> If we're happy with this, please # it.

Done

**Pull request merged**